File: 0169/0169.c
// Author: washing
// DateTime: 2021/1/26 18:50
// File: 0169.c
// Desc: 

int majorityElement(int* nums, int numsSize){
    int a = 0;
    int c = 0;
    for (int i=0; i<numsSize; i++){
        if (c==0){
            a = nums[i];
            c++;
        }else if (a==nums[i]){
            c++;
        }else{
            c--;
        }
    }
    return a;
}

File: 0169/0169.scala
// Author: washing
// DateTime: 2021/1/26 18:48
// File: 0169.scala
// Desc: 

object Solution {
    def majorityElement(nums: Array[Int]): Int = {
        var a=0;
        var c=0;
        
        for (n <- nums) {
            if (c==0){
                a = n
                c += 1
            } else if (a==n){
                c += 1
            } else {
                c -= 1
            }
        }
        return a
    }
}